frontend/src/components/MultiselectDropdown.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react";
+import "../MultiselectDropdown.css";
+import dropdownIcon from "../assets/icons/dropdown-arrow.png"; // Local image
+
+interface MultiselectDropdownProps {
+    options: string[];
+    label: string;
+    selectedOptions: string[];
+    setSelectedOptions: (value: string[]) => void;
```

Comment:
It seems that the type defined for `setSelectedOptions` does not match its intended use, you need something like this:
```suggestion
    setSelectedOptions: (fn: (prev: string[]) => string[]) => void;
```
or
```suggestion
    setSelectedOptions: Dispatch<SetStateAction<string[]>>;
```